### PR TITLE
[NFC-395] Add delegate for tracking NFC detection

### PR DESCRIPTION
**Pull request by laurageiger-mohemian**
Issue [NFC-395](https://airside.atlassian.net/browse/NFC-395)

Created delegate for passing detection of NFC tag.

[NFC-395]: https://airside.atlassian.net/browse/NFC-395?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by damiankobialka-mohemian — APPROVED** [review]

File: Sources/NFCPassportReader/PassportReader.swift [hunk 1]
```diff
@@ -285,6 +291,7 @@ extension PassportReader {
     func doBACAuthentication(tagReader : TagReader) async throws {
         self.currentlyReadingDataGroup = nil
         
+        trackingDelegate?.detectNFCTag()
```
Comment (PatrykWinowski-Moh) [inline Sources/NFCPassportReader/PassportReader.swift:294]:
```suggestion
        trackingDelegate?.nfcTagDetected()
```

File: Sources/NFCPassportReader/PassportReader.swift [hunk 2]
```diff
@@ -12,11 +12,17 @@ import Foundation
 import UIKit
 import CoreNFC
 
+@available(iOS 15, *)
+public protocol PassportReaderDelegate: AnyObject {
+    func detectNFCTag()
```
Comment (PatrykWinowski-Moh) [inline Sources/NFCPassportReader/PassportReader.swift:17]:
```suggestion
    func nfcTagDetected()
```

**Review by PatrykWinowski-Moh — COMMENTED** [review]
It's good when delegate method's names clearly suggest that these are reactive (callback) methods rather than imperative.

File: Sources/NFCPassportReader/PassportReader.swift (hunk 2 above)
Comment (laurageiger-mohemian) [reply Sources/NFCPassportReader/PassportReader.swift:17]: Thanks @PatrykWinowski-Moh! 💯

**Review by PatrykWinowski-Moh — APPROVED** [review]
👌

File: Sources/NFCPassportReader/PassportReader.swift [hunk 3]
```diff
@@ -12,11 +12,17 @@ import Foundation
 import UIKit
 import CoreNFC
 
+@available(iOS 15, *)
+public protocol PassportReaderDelegate: AnyObject {
```
Comment (daniel-moh) [inline Sources/NFCPassportReader/PassportReader.swift:16]: What would you think about:

```suggestion
public protocol PassportReaderTrackingDelegate: AnyObject {
```

I feel like that more clearly suggests what it is about. From the name `PassportReaderDelegate` one could assume it is a general delegate that also communicates errors and results.

File: Sources/NFCPassportReader/PassportReader.swift [hunk 4]
```diff
@@ -12,11 +12,17 @@ import Foundation
...
+    func nfcTagDetected()
+}
+
 @available(iOS 15, *)
 public class PassportReader : NSObject {
     private typealias NFCCheckedContinuation = CheckedContinuation<NFCPassportModel, Error>
     private var nfcContinuation: NFCCheckedContinuation?
 
+    public weak var trackingDelegate: PassportReaderDelegate?
```
Comment (daniel-moh) [inline Sources/NFCPassportReader/PassportReader.swift:25]:
```suggestion
    public weak var trackingDelegate: PassportReaderTrackingDelegate?
```

The "tracking" in the name is even already there  ☝️ 😄

File: Sources/NFCPassportReader/PassportReader.swift (hunk 3 above)
Comment (laurageiger-mohemian) [reply Sources/NFCPassportReader/PassportReader.swift:16]: Agree! Naming it `PassportReaderTrackingDelegate` makes more sense

**Pull request merged**